src/AssocValue.php

```diff
@@ -1,146 +1,171 @@
-<?php
+<?php declare(strict_types=1);
 
 namespace GW\Value;
 
+use BadMethodCallException;
 use IteratorAggregate;
 use ArrayAccess;
 
+/**
+ * @template TKey
+ * @template TValue
+ * @extends Collection<TValue>
+ * @extends IteratorAggregate<TKey, TValue>
+ * @extends ArrayAccess<TKey, TValue>
+ */
 interface AssocValue extends Value, Collection, IteratorAggregate, ArrayAccess
 {
-    // Collection
     /**
-     * @param callable $callback function(mixed $value): void
-     * @return AssocValue
+     * @phpstan-param callable(TValue $value, TKey $key=):void $callback
```

Comment:
Here another one. I'm start believing that it is something I didn't know about... ;)